tyrian/js/src/main/scala/tyrian/Cmd.scala

```diff
@@ -69,6 +70,14 @@ object Cmd:
     def apply[F[_], A, Msg](task: F[A])(toMessage: A => Msg): Run[F, A, Msg] =
       Run(task, toMessage)
 
+  /** Builds and runs a task that needs a Dispatcher */
+  final case class BuildRun[F[_], A, Msg](
+      build: Dispatcher[F] => F[A],
+      toMsg: A => Msg
+  ) extends Cmd[F, Msg]:
+    def map[OtherMsg](f: Msg => OtherMsg): BuildRun[F, A, OtherMsg] =
+      BuildRun(build, toMsg andThen f)
```

Comment:
I'm not hugely keen on the name `BuildRun` (what does that mean? why does is speak the need for a `Dispatcher`?), but the implementation is interesting, I wouldn't have thought of that.

I need to have a think about this a bit. I feel like its opening a can of worms but I'm not sure how big the can is yet. :smile:

Reply:
Teehee... Yes, I don't like it either. It's just that I liked the ten names I thought of before even less. :)

"It builds a `Run`." I'll think about it some more.